NuRadioMC/EvtGen/generator.py

```diff
@@ -1578,14 +1579,78 @@ def split_hdf5_input_file(input_filename, output_filename, number_of_events_per_
         the number of events per file
     """
     fin = h5py.File(input_filename, 'r')
-    data_sets = {}
     attributes = {}
-    for key, value in iteritems(fin):
-        if isinstance(value, h5py.Dataset):  # the loop is also over potential subgroupu that we don't want to consider here
-            data_sets[key] = np.array(value)
+    data = {}
+    groups = {}
+    n_groups = {}
+    n_data = {}
+    group_attrs = {}
+
+    for key in fin:
+        if isinstance(fin[key], h5py._hl.group.Group):
+            groups[key] = {}
+            if(key not in n_groups):
+                n_groups[key] = {}
+            for key2 in fin[key]:
+                groups[key][key2] = fin[key][key2][...]
+                if(key2 not in n_groups[key]):
+                    n_groups[key][key2] = 0
+                n_groups[key][key2] += len(groups[key][key2])
+            if(key not in group_attrs):
+                group_attrs[key] = {}
+                for key2 in fin[key].attrs:
+                    group_attrs[key][key2] = fin[key].attrs[key2]
+            else:
+                for key2 in fin[key].attrs:
+                    if(not np.all(group_attrs[key][key2] == fin[key].attrs[key2])):
+                        logger.warning(f"attribute {key2} of group {key} of file {input_filename} are different ({group_attrs[key][key2]} vs. {fin[key].attrs[key2]}. Using attribute value of first file, but you have been warned!")
+        else:
+            data[key] = fin[key][...]
+            if(key not in n_data):
+                n_data[key] = 0
+            n_data[key] += len(data[key])
+
     for key, value in iteritems(fin.attrs):
         attributes[key] = value
-
+#     logger.info(f"setting number of events from {attributes['n_events']} to the actual number of events in this file {len(data_sets['event_ids'])}")
     fin.close()
 
-    write_events_to_hdf5(output_filename, data_sets, attributes, n_events_per_file=number_of_events_per_file)
+    n_events = len(data['event_ids'])
+    logger.info("saving {} events in total".format(n_events))
+    total_number_of_events = attributes['n_events']
+
+    n_files = math.ceil(n_events / number_of_events_per_file)
+    for iFile in range(n_files):
+        filename2 = output_filename + ".part{:04}".format(iFile)
+        logger.debug(f"saving file {iFile} with {number_of_events_per_file} events to {filename2}")
+        fout = h5py.File(filename2, 'w')
+        for key, value in attributes.items():
+            fout.attrs[key] = value
+        fout.attrs['total_number_of_events'] = total_number_of_events / n_files
```

Comment:
Is this correct? Isn't this a float version of fout.attrs['n_events']?

Reply:
yes, after splitting files, the original total number of event of this file might not be an int.